### PR TITLE
Fix output generation for device validation reports

### PR DESCRIPTION
**Pull request by grafnu**
Things in validator/trace/simple/ can be ignored (trace input files)
Things in validator/trace/simple.out/ are examples of what the output produced is, so should be checked for completeness.
Other code bits deserve code-review as per normal.

### CODE REVIEW

File: validator/src/main/java/com/google/daq/mqtt/validator/Validator.java [hunk 1]
```diff
@@ -538,6 +534,11 @@ private void sendValidationResult(Map<String, String> origAttributes, Object mes
       ReportingDevice reportingDevice) {
     try {
       ValidationEvent validationEvent = makeValidationEvent();
+      validationEvent.status = reportingDevice.getErrorStatus();
```
Comment (johnrandolph) [inline validator/src/main/java/com/google/daq/mqtt/validator/Validator.java:537]: Should makeValidationEvent() help with some of the building below?

**Comment by grafnu** [comment]
The function creates the basic common bits, but then it's filled in quite
differently for the different uses -- I'm all for generalization, but not
quite sure how to do it in this case!

File: validator/src/main/java/com/google/daq/mqtt/validator/Validator.java (hunk 1 above)
Comment (grafnu) [reply validator/src/main/java/com/google/daq/mqtt/validator/Validator.java:530]: The function creates the basic common bits, but then it's filled in quite
differently for the different uses -- I'm all for generalization, but not
quite sure how to do it in this case!

File: validator/src/main/java/com/google/daq/mqtt/validator/ReportingDevice.java [hunk 2]
```diff
@@ -200,35 +208,57 @@ public Entry getErrorStatus() {
     if (errors.isEmpty()) {
       return null;
     }
+    return errors.size() == 1 ? makeEntry(errors.get(0)) : makeCompoundEntry(errors);
+  }
+
+  private Entry makeCompoundEntry(List<Exception> erx) {
```
Comment (johnrandolph) [inline validator/src/main/java/com/google/daq/mqtt/validator/ReportingDevice.java:214]: Could we have docstrings that give example input/output here? github doesn't lend itself to jumping around very quickly to piece it together.

File: validator/src/main/java/com/google/daq/mqtt/validator/Validator.java (hunk 1 above)
Comment (johnrandolph) [reply validator/src/main/java/com/google/daq/mqtt/validator/Validator.java:530]: ok

**Review by johnrandolph — APPROVED** [review]
I think at some point we need a util class that has all these general purpose functions like exception translators etc? But for now lgtm

**Pull request merged**